dmutils/email/dm_mailchimp.py

```diff
@@ -130,6 +129,6 @@ def get_email_addresses_from_list(self, list_id, pagination_size=1000):
             if not member_data.get("members", None):
                 break
             offset += pagination_size
-            email_addresses.extend(member["email_address"] for member in member_data["members"])
 
-        return email_addresses
+            for member in member_data["members"]:
+                yield member['email_address']
```

Comment:
FWIW if you wanted to show off you could do this as a `yield from (member['email_address'] for member in member_data["members"])`

Reply:
Aha. I had a feeling something like this existed but didn't follow through on that feeling to actually look it up.

Thanks